### ISSUE TITLE
largest-series-product odd expected return value of 1 from "", 0 input

### ISSUE THREAD

**Issue by matthewfischer** [exercism/xgo#349]
I can't understand the logic for these two examples. I believe they are both clearly errors. The first one passes the example code but I believe it's an implementation detail. The second one is clearly an error, but I believe the first one is at best undefined.

Input: "", 0 -> 1
Input: "", 1 -> -1, error

The 0th largest-series-product of nothing is at best undefined and I considered it an error in my code, what's the logic that gets a value of 1 out of that?

**Comment by petertseng** [comment]
How many substrings of length zero are in the empty string? One, the empty string. What is the product of the empty string? It's the empty product, 1.

Comments at https://github.com/exercism/x-common/pull/170

**Comment by petertseng** [comment]
Despite it mathematically making sense (being identity under multiplication), there's not much indication that it has to be that way, though. Maybe needs to be specified.

**Comment by matthewfischer** [comment]
Thanks for the reference. I'm with you on the empty string part. Does the latter part mean that nothing times nothing is 1?

**Comment by petertseng** [comment]
> Does the latter part mean that nothing times nothing is 1?

I've got two ways of looking at this.

If I were to have the number 6, and multiply it by two other numbers, it's as if I multiplied 6 by the product of the two other numbers. So, if I multiplied 6 by no other numbers, it's as if I multiplied 6 by 1.

Alternatively, if I were adding numbers up in a loop, I'd probably do something like:

```go
sum := 0
for _, num in range nums {
    sum += num
}
```

And if I were multiplying?

```go
product := 1
for _, num in range nums {
    product *= num
}
```

In the sum case, I definitely have to start from 0 for the final result to be correct. And in the product case, I definitely have to start from 1 for the final result to be correct too. and so we see that the product would remain 1 if `nums` happened to be empty.

---

All this aside, it still might be useful to say something about that in the README
